Sources_v3/StreamChat/ChatClient.swift

```diff
@@ -124,26 +120,24 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
     
     /// The `WebSocketClient` instance `Client` uses to communicate with Stream WS servers.
     lazy var webSocketClient: WebSocketClient? = {
-        // Create a connection request
-        let webSocketEndpoint: Endpoint<EmptyResponse> = .webSocketConnect(
-            userId: self.currentUserId,
-            name: nil,
-            imageURL: nil,
-            extraData: nil as ExtraData.User?
-        )
-        
         var encoder = environment.requestEncoderBuilder(config.baseURL.webSocketBaseURL, config.apiKey)
         encoder.connectionDetailsProviderDelegate = self
                 
         // Create a WebSocketClient.
         let webSocketClient = environment.webSocketClientBuilder(
-            webSocketEndpoint,
             urlSessionConfiguration,
             encoder,
             EventDecoder<ExtraData>(),
             eventNotificationCenter,
             internetConnection
         )
+
+        if let currentUserId = currentUserId {
+            webSocketClient?.connectEndpoint = .webSocketConnect(
+                userId: currentUserId,
+                extraData: nil as ExtraData.User?
+            )
+        }
```

Comment:
This has to be handled differently. We shouldn't even try to connect when we don't have user id.

Reply:
moved to `prepareEnvironment` where user id is known